provision-osx/tasks/configure_buildfarm_node.yml

```diff
@@ -0,0 +1,98 @@
+    connection: local
+    get_url:
+      url: "{{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }}/jnlpJars/jenkins-cli.jar"
+      dest: /tmp/jenkins-cli.jar
+      force: true
+      mode: 0775
+
+  -
+    block:
+    -
+      name: Retrieve provided private key
+      shell: "cat {{ credential_private_key_path }}"
+      register: private_key
+
+    -
+      name: Retrieve provided public key
+      shell: "cat {{ credential_public_key_path }}"
+      register: public_key
+
+    -
+      name: Set credential_private_key and credential_passphrase
+      set_fact:
+        credential_private_key: "{{ private_key.stdout }}"
+        credential_public_key: "{{ public_key.stdout }}"
+        credentials_provided: true
+
+    -
+      name: Create credential config from template
+      template:
+        src: create-credential.xml.j2
+        dest: "/tmp/create-credential.xml"
+        force: yes
+
+    -
+      name: Create credential
+      shell: cat /tmp/create-credential.xml | java -jar /tmp/jenkins-cli.jar -remoting -s {{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }} create-credentials-by-xml system::system::jenkins _
+      args:
+        executable: /bin/bash
+      register: create_credential
+      failed_when: create_credential.rc != 0 and create_credential.stderr.find('No change') == -1
+      changed_when: create_credential.rc == 0
+
+    -
+      name: Update credential
+      shell: "cat /tmp/create-credential.xml | java -jar /tmp/jenkins-cli.jar -remoting -s {{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }} update-credentials-by-xml system::system::jenkins _ {{ buildfarm_credential_id }}"
+      args:
+        executable: /bin/bash
+      register: update_credential
+      when: create_credential.rc != 0 and create_credential.stderr.find('No change') >= 0
+    run_once: true
+    connection: local
+
+  -
+    block:
+    -
+      name: Create node config from template
+      template:
+        src: create-node.xml.j2
+        dest: "/tmp/create-node.xml"
+        force: yes
+    -
+      name: Create node
+      shell: cat /tmp/create-node.xml | java -jar /tmp/jenkins-cli.jar -remoting -s {{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }} create-node
+      args:
+        executable: /bin/bash
+      register: create_node
+      failed_when: create_node.rc != 0 and create_node.stderr.find('already exists') == -1
+      changed_when: create_node.rc == 0
+
+    -
+      name: Update node
+      shell: "cat /tmp/create-node.xml | java -jar /tmp/jenkins-cli.jar -remoting -s {{ jenkins_route_protocol | default('https') }}://{{ route_output.stdout }} update-node '{{ buildfarm_node_name }}'"
+      args:
+        executable: /bin/bash
+      register: create_node
+      when: create_node.rc != 0 and create_node.stderr.find('already exists') >= 0
+    connection: local
+
+  -
+    authorized_key:
```

Comment:
should this step be done before creating node, given that Jenkins will try to launch the node immediately after creating the node?